### ISSUE TITLE
Migrating package examples

### ISSUE THREAD

**Issue by timsneath** [dart-lang/site-www#2776]
## Page

https://dart.dev/tools/pub/create-packages (new section)

## Notes

One note from my last-mile experience: there's no mention in the docs about migrating examples / samples, yet they're quite likely to have a separate `pubspec.yaml`. 

That introduces a number of (unanswered) questions. For example, what if an example doesn't have all its dependencies migrated? Is it better to not start the migration? Start the migration but leave the example unmigrated? Migrate the example and just ignore the complaints about its dependencies? 

/cc @kevmoo

**Comment by kevmoo** [comment]
We absolutely want to make sure the example still runs.

The only way to do that (without requiring a user to pass flags) is to have the example flagged with `// @dart=2.9`

Either way, we should tell folks to migrate their examples, but if they cannot be run strongly, the example should have some screaming comments at the top telling folks such

**Comment by atsansone** [comment]
@kevmoo : Is this still an active concern?

**Comment by kevmoo** [comment]
CC @mit-mit who looks after the package ecosystem. I'm guessing when analysis rolls over to Dart 3.0, existing examples will break. Not sure about guidance, though.

**Comment by mit-mit** [comment]
Yes, this is still a valid concern for any package that needs to be migrated to either 2.12 or 3.0
